exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,108 @@
 )
 
+const service = "grafana"
+const check = "prometheus-appstudio-ds"
+var getInClusterConfig = config.GetConfigOrDie
+var getNewForConfig = kubernetes.NewForConfigOrDie
+
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxUp *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxUp: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_up",
+			Help: "Availability of the Konflux default grafana datasource",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxUp.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	var availability float64
+
+	availability = CheckDataSourceExist(GetDataSources(GetGrafanaResource(NewKubeClient())), check)
+	e.konfluxUp.WithLabelValues(service, check).Set(availability)
+	e.konfluxUp.Collect(ch)
+}
+
+// get the grafna resource as a map
+func GetGrafanaResource(clientset *kubernetes.Clientset) map[string]interface{} {
+	data, err := clientset.RESTClient().
+		Get().
+		AbsPath("/apis/grafana.integreatly.org/v1beta1").
+		Namespace("appstudio-grafana").
+		Resource("grafanas").
+		Name("grafana-oauth").
+		DoRaw(context.TODO())
+	var grafanaResource map[string]interface{}
+	err = json.Unmarshal(data, &grafanaResource)
+	if err != nil {
+		fmt.Printf("Error getting resource: %v\n", err)
+		os.Exit(1)
+	}
+
+	return grafanaResource
+}
+
+// get datasources from grafana resource
+func GetDataSources(grafanaResource map[string]interface{}) []string {
+	// return empty string slice if datasources are not defined
+	if grafanaResource["status"].(map[string]any)["datasources"] == nil {
+		return make([]string, 0)
+	}
+	datasourcesIfc := grafanaResource["status"].(map[string]any)["datasources"].([]interface{})
+	datasources := make([]string, len(datasourcesIfc))
+	for i, v := range datasourcesIfc {
+		datasources[i] = v.(string)
+	}
+	return datasources
+}
+
+// check if datasource exists, return 1 if yes, 0 if not
+func CheckDataSourceExist(datasources []string, dsToCheck string) float64 {
+	for _, datasource := range datasources {
+		if strings.Contains(datasource, dsToCheck) {
```

Comment:
What's the reason for using `Contains` rather than checking if the strings are equal?

Reply:
I think we should be able to replace this entire loop a call to `slices.Contains`.
See here: https://stackoverflow.com/a/71181131

Reply:
Here the datasource we are getting is `appstudio-grafana/prometheus-appstudio-ds/b80cfcd5-3010-4e6a-be44-6843e0fef2d8` so when we are checking for `prometheus-appstudio-ds` `Contains` returns true. Also, `slices.Contains` is checking for exact match in the string so it's not working.

Reply:
> Here the datasource we are getting is appstudio-grafana/prometheus-appstudio-ds/b80cfcd5-3010-4e6a-be44-6843e0fef2d8

We should let the test case reflect that so the structure of the inputs is clear.

> Also, slices.Contains is checking for exact match in the string so it's not working.

If we cannot use exact match, then we should narrow down the chances of false positives. Either use some sort of regex or use the expected format of the datasource that is returning to make the chances of false match slimmer.

Reply:
Updated test case and using regex for matching.

Reply:
Updated back to `strings.Contains`  again